### DISCUSSION TITLE
Is it possible to change the active area of the map?

### DISCUSSION THREAD

**Discussion by FKonyali** [visgl/react-google-maps#251] [Ideas]
<img src="https://github.com/visgl/react-google-maps/assets/16213100/ee29fb57-f1ed-4539-9e11-eeaea2bd0baa" width="300" />

I am sharing screenshots as examples. Let's imagine that we have a map that covers the full screen as seen above. There is also an action sheet that opens when you click on the markers.

<img src="https://github.com/visgl/react-google-maps/assets/16213100/c3c762e1-f58f-4a81-91dc-651995d1b0c2" width="300" />

In the second image, it can be seen that the active area of the map has become smaller. In the marker, this appears to be centered in the active area.

I use the "react-native-maps" library on React Native. This library solves this active area need with mapPadding props. These props take the values top, left, right and bottom. In this way, the active area and related tearms links remain above. I think props like this can be added to the library. What do you think?

**Comment by usefulthink** [comment]
Unfortunately, that feature is missing from the Maps JavaScript API. 
For things like such an action-sheet it's generally ok to cover the attribution and google-logo if it's only temporary, so the map can remain it's original size behind the action-sheet. The only problem with that is that you need to do some math if you want to position the map such that some location is centered in the visible portion of the viewport. The alternative is to resize the map-element itself, which also works relatively well.
